core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/QueryEvaluationContext.java

```diff
@@ -33,26 +33,31 @@ public interface QueryEvaluationContext {
 	public class Minimal implements QueryEvaluationContext {
 		public Minimal(Literal now, Dataset dataset) {
 			super();
-			this.now = now;
+			this.nowL = now;
+			// This is valid because the now field will never be read.
+			this.now = 0L;
 			this.dataset = dataset;
 		}
 
 		public Minimal(Dataset dataset) {
+			this.now = System.currentTimeMillis();
 			this.dataset = dataset;
 		}
```

Comment:
Normally I would suggest that the second constructor should just invoke the first one:

```java
public Minimal(Dataset dataset) {
    this(System.currentTimeMillis(), dataset);
}
```

But what's going on on line 38 confuses me. Why are you resetting the `now` value to 0 (and why are you doing it in the first constructor but not in the second one)? And if the `now` field is "never read", why even have one?

Reply:
@jeenbroekstra  I added comments about what is going on.

Reply:
Going with the simpler varhandle approach.